docs/native-modules-ios.md

```diff
@@ -1,442 +1,466 @@
 
 ```jsx
-CalendarManager.addEvent(
-  'Birthday Party',
-  '4 Privet Drive, Surrey',
-  date.toISOString()
-); // passing date as ISO-8601 string
+import { NativeModules } from 'react-native';
 ```
 
-And both values would get converted correctly to the native `NSDate`. A bad value, like an `Array`, would generate a helpful "RedBox" error message.
+You can then access the `CalendarModule` native module off of `NativeModules`.
 
-As `CalendarManager.addEvent` method gets more and more complex, the number of arguments will grow. Some of them might be optional. In this case it's worth considering changing the API a little bit to accept a dictionary of event attributes, like this:
+```jsx
+const { CalendarModule } = NativeModules;
+```
 
-```objectivec
-#import <React/RCTConvert.h>
+Now that you have the CalendarModule native module available, you can invoke your native method `createCalendarEvent`. Below it is added to the `onPress` method in `NewModuleButton`:
 
-RCT_EXPORT_METHOD(addEvent:(NSString *)name details:(NSDictionary *)details)
-{
-  NSString *location = [RCTConvert NSString:details[@"location"]];
-  NSDate *time = [RCTConvert NSDate:details[@"time"]];
-  ...
-}
+```jsx
+const onPress = () => {
+  CalendarModule.createCalendarEvent('testName', 'testLocation');
+};
 ```
 
-and call it from JavaScript:
+The final step is to rebuild the React Native app so that you can have the latest native code (with your new native module!) available. In your command line, where the react native application is located, run the following :
 
 ```jsx
-CalendarManager.addEvent('Birthday Party', {
-  location: '4 Privet Drive, Surrey',
-  time: date.getTime(),
-  description: '...'
-});
+react-native run-ios
 ```
 
-> **NOTE**: About array and map
->
-> Objective-C doesn't provide any guarantees about the types of values in these structures. Your native module might expect an array of strings, but if JavaScript calls your method with an array containing numbers and strings, you'll get an `NSArray` containing a mix of `NSNumber` and `NSString`. For arrays, `RCTConvert` provides some typed collections you can use in your method declaration, such as `NSStringArray`, or `UIColorArray`. For maps, it is the developer's responsibility to check the value types individually by manually calling `RCTConvert` helper methods.
+### Building as You Iterate
 
-## Callbacks
+As you work through these guides and iterate on your native module, you will need to do a native rebuild of your application to access your most recent changes from Javascript. This is because the code that you are writing sits within the native part of your application. While React Native’s metro bundler can watch for changes in Javascript and rebuild JS bundle on the fly for you, it will not do so for native code. So if you want to test your latest native changes you need to rebuild by using the `react-native run-ios` command.
```

Comment:
Again, please change the run command to `npx react-native run-ios`.

Reply:
done